projects/09/Snake/Game.jack

```diff
@@ -0,0 +1,153 @@
+class Game {
+	field int screenWidth, screenHeight, topLimit, bottomLimit, rightLimit, leftLimit, borderSize;
+	field int score;
+	field Snake snake;
+	field Random rand;
+	field boolean foodExists;
+	field Food food;
+
+    /** Constructs a new Game object. */
+    constructor Game new(int screenWidthIn, int screenHeightIn, int borderSizeIn) {
+		let borderSize = borderSizeIn;
+		let screenWidth = screenWidthIn;
+		let screenHeight = screenHeightIn;
+        let topLimit = borderSize;
+        let bottomLimit = screenHeightIn - borderSize;
+        let rightLimit = screenWidthIn - borderSize;
+        let leftLimit = borderSize;
+		let score = 0;
+		do createSnake();
+        return this;
+    }
+
+    /** Deallocates the object's memory. */
+    method void dispose() {
+		do snake.dispose();
+		do rand.dispose();
+		do food.dispose();
+        do Memory.deAlloc(this);
+        return;
+    }
+	
+	method void createSnake() {
+		let snake = Snake.new(15);
+		do snake.initBasicSnake(200, 120);
+		return;
+	}
+	
+	method void createFood() {
+		var int x, y, radius;
+		let radius = 5;
+		let x = rand.randRange(rightLimit - leftLimit - radius) + leftLimit + radius;
+		let y = rand.randRange(bottomLimit - topLimit - radius) + topLimit + radius;
+		let food = Food.new(x, y, radius);
+		let foodExists = true;
+		do food.draw();
+		return;
+	}
+	
+	/** Returns true if the snake collides with the food */
+	method boolean detectFoodCollison() {
+		if (snake.doesCollide(food.getY() - food.getRadius() - 5, food.getY() + food.getRadius() + 5, food.getX() - food.getRadius() - 5, food.getX() + food.getRadius() + 5)) { return true; }
```

Comment:
what does the 5 mean?
+ break into multiple lines

Reply:
Simple padding that makes the game easier to play (The player will get the food even if not exactly on it).